File: 145-binary-tree-postorder-traversal/145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> ans = new ArrayList<>();
    public void postorderTraversalUtil(TreeNode root) {
        if (root != null) {
            postorderTraversalUtil(root.left);
            postorderTraversalUtil(root.right);
            ans.add(root.val);
        }
    }
    public List<Integer> postorderTraversal(TreeNode root) {
        postorderTraversalUtil(root);
        return ans;
    }
}

File: 145-binary-tree-postorder-traversal/145-binary-tree-postorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
let ans = []
var postorderTraversalUtil = function(root) {
    if (root) {
        postorderTraversalUtil(root.left)
        postorderTraversalUtil(root.right)
        ans.push(root.val)
    }
}
var postorderTraversal = function(root) {
    ans.length = 0
    postorderTraversalUtil(root)
    return ans
}